app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -159,6 +163,42 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         components.appStartupTelemetry.onFenixApplicationOnCreate()
     }
 
+    private fun restoreBrowsingSession() = GlobalScope.launch(Dispatchers.Main) {
+        val store = components.core.store
+        val sessionStorage = components.core.sessionStorage
+
+        components.useCases.tabsUseCases.restore(sessionStorage)
+
+        // Now that we have restored our previous state (if there's one) let's setup auto saving the state while
+        // the app is used.
+        sessionStorage.autoSave(store)
+            .periodicallyInForeground(interval = 30, unit = TimeUnit.SECONDS)
+            .whenGoingToBackground()
+            .whenSessionsChange()
+
+        // TODO: Move into use case?
+        store.dispatch(RestoreCompleteAction)
+    }
+
+    private fun removeTimedOutTabs() {
```

Comment:
Let's file a follow-up to cover this with a test. I didn't have one before it seems ...